ttnn/cpp/ttnn/operations/normalization/layernorm/device/multi_core/layernorm_op_multi_core.cpp

```diff
@@ -845,6 +849,31 @@ operation::ProgramWithCallbacks layernorm_multi_core_sharded(
             num_cores_y_mcast = num_cores_y;
         }
     }
+    auto applyStartOffset = [](const CoreRangeSet& input_set, const CoreCoord& grid_offset) -> CoreRangeSet {
+        if (input_set.empty()) {
+            return input_set;
+        }
+
+        std::vector<CoreRange> new_ranges;
+        new_ranges.reserve(input_set.size());
+
+        for (const CoreRange& range : input_set.ranges()) {
+            CoreCoord new_start = {range.start_coord.x + grid_offset.x, range.start_coord.y + grid_offset.y};
+            CoreCoord new_end = {range.end_coord.x + grid_offset.x, range.end_coord.y + grid_offset.y};
+            new_ranges.emplace_back(new_start, new_end);
+        }
+
+        return CoreRangeSet(std::move(new_ranges));
+    };
+    if (grid_offset.has_value()) {
+        start_core = {start_core.x + grid_offset.value().x, start_core.y + grid_offset.value().y};
+        sender_cores = {
+            {sender_cores.start_coord.x + start_core.x, sender_cores.start_coord.y + start_core.y},
+            {sender_cores.end_coord.x + start_core.x, sender_cores.end_coord.y + start_core.y}};
+        all_to_all_cores = applyStartOffset(all_to_all_cores, grid_offset.value());
+        all_to_all_workers_except_sender = applyStartOffset(all_to_all_workers_except_sender, grid_offset.value());
+        not_all_to_all_workers = applyStartOffset(not_all_to_all_workers, grid_offset.value());
+    }
```

Comment:
Is this necessary? Can we not set the original start_core from {0, 0} to the actual start core in the code above?

Reply:
No, actually. Most of these core range sets were created using the num_cores_to_core_range_set() API, which assumes that the core grid always starts at (0,0) and iterates through the given start_core from there. In addition, there are tons of hard-coded conditionals for getting start and end ranges, which were hard to decode. Therefore, I thought I would just shift the core grid once to avoid going through all the conditions.

Reply:
tldr: start_core is not a core_grid offset. It is the start of corerange in grid ((0,0)->(grid_size.x, grid_size.y)).
To do above in more cleaner way we need a new api num_cores_to_corerangeset which also takes core_grid start

Reply:
Okay, this is fine for now. We can revisit later if we can make it cleaner

Reply:
Yeah agree, there is scope to make it cleaner.